src/js/common/schemaform/actions.js

```diff
@@ -81,3 +114,180 @@ export function submitForm(formConfig, form) {
+ *                                version of the form the data was saved with
+ *                                is different from the current version.
+ */
+export function loadFormData(formId, migrations) {
+  // TODO: Test if the form is still saved after submission.
+  return dispatch => {
+    const userToken = sessionStorage.userToken;
+    // If we don't have a userToken, fail safely
+    if (!userToken) {
+      dispatch(setLoaded('no-auth')); // Shouldn't get here, but just in case
+      return;
+    }
+
+    // Update UI while we're waiting for the API
+    dispatch(setLoaded('pending'));
+
+    // Query the api
+    fetch(`${environment.API_URL}/v0/in_progress_forms/${formId}`, {
+      // TODO: These headers should work, but trigger an api error right now
+      headers: {
+        'Content-Type': 'application/json',
+        // 'X-Key-Inflection': 'camel',
+        Authorization: `Token token=${userToken}`
+      },
+    }).then((res) => {
+      // console.log('res', res);
+      if (res.ok) {
+        return res.json();
+      }
+
+      let status = 'failure';
+      if (res.status === 401) {
+        // TODO: If they've sat on the page long enough for their token to expire
+        //  and try to load, tell them their session expired and they need to log
+        //  back in and try again.
+        status = 'no-auth';
+      } else if (res.status === 404) {
+        status = 'not-found';
+      }
+      return Promise.reject(status);
+    }).then((resBody) => {  // eslint-disable-line consistent-return
+      // Just in case something funny happens where the json returned isn't an object as expected
+      if (typeof resBody !== 'object') {
+        return Promise.reject('invalid-data');
+      }
+
+      // If an empty object is returned, throw a not-found
+      // TODO: When / if we return a 404 for applications that don't exist, remove this
+      if (Object.keys(resBody).length === 0) {
+        return Promise.reject('not-found');
+      }
+
+      // If we've made it this far, we've got valid form
+
+      let formData;
+      try {
+        // NOTE: This may change to be migrated in the back end before sent over
+        formData = getUpdatedFormData(resBody.form_data, resBody.metadata.version, migrations);
+        // formData = getUpdatedFormData(resBody.formData, resBody.metadata.version, migrations);
+      } catch (e) {
+        // TODO: Log e.message somewhere; it's the reason the data couldn't be
+        //  transformed. Sentry error?
+        return Promise.reject('invalid-data');
+      }
+      // Set the data in the redux store
+      // NOTE: Until we get the api for the list of filled forms, we're using this
+      //  function to see if the form has been filled in, so this is setting the
+      //  data in a separate place to be pulled in when we _actually_ want to load
+      //  the formData.
+      // dispatch(setData(formData));
+      dispatch(setLoadedData({ formData, metadata: resBody.metadata }));
```

Comment:
In the reducer for this action, can you set the data like you're doing now and also set the status to 'success?' Instead of firing two actions?